src/services/base/labeling.ts

```diff
@@ -0,0 +1,164 @@
+}
+
+export function generateAccessLink(projectId: string, options: {
+    type: string, id: string,
+}, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/generate-access-link`;
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(options)));
+}
+
+export function removeAccessLink(projectId: string, linkId: string, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/remove-access-link`;
+    jsonFetchWrapper(finalUrl, FetchType.DELETE, onResult, JSON.stringify({ "value": linkId }));
+}
+
+export function lockAccessLink(projectId: string, options: {
+    linkId: string, lockState: boolean
+}, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/lock-access-link`;
+    jsonFetchWrapper(finalUrl, FetchType.PUT, onResult, JSON.stringify(convertCamelToSnakeCase(options)));
+}
+
+export function addClassificationLabels(projectId: string, recordId: string, labelingTaskId: string, labelId: string, asGoldStar: boolean, sourceId: string, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/add-classification-labels`;
+    const body = {
+        recordId,
+        labelingTaskId,
+        labelId,
+        asGoldStar,
+        sourceId
+    }
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(body)));
+}
+
+export function addExtractionLabel(projectId: string, recordId: string, labelingTaskId: string, tokenStartIndex: number, tokenEndIndex: number, value: string, labelId: string, asGoldStar: boolean, sourceId: string, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/add-extraction-label`;
+    const body = {
+        recordId,
+        labelingTaskId,
+        tokenStartIndex,
+        tokenEndIndex,
+        value,
+        labelId,
+        asGoldStar,
+        sourceId
+    }
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(body)));
+}
+
+export function setGoldStar(projectId: string, recordId: string, labelingTaskId: string, goldUserId: string, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/set-gold-star`;
+    const body = {
+        recordId,
+        labelingTaskId,
+        goldUserId
+    }
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(body)));
+}
+
+export function removeGoldStar(projectId: string, recordId: string, labelingTaskId: string, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/remove-gold-star`;
+    const body = {
+        recordId,
+        labelingTaskId
+    }
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(body)));
+}
+
+export function updateLabelingTask(projectId: string, options: {
```

Comment:
everything from here to the end should go in a separate file in my opinion since they are not part of the labeling. They are part of the project settings but i think the logic can be split into separate file so it's not to much code in one file